rpcclient/wallet.go

```diff
@@ -2608,12 +2608,128 @@ func (c *Client) GetWalletInfo() (*btcjson.GetWalletInfoResult, error) {
+	return c.sendCmd(btcjson.NewBackupWalletCmd(destination))
+}
+
+// BackupWallet safely copies current wallet file to destination, which can
+// be a directory or a path with filename
+func (c *Client) BackupWallet(destination string) error {
+	return c.BackupWalletAsync(destination).Receive()
+}
+
+// FutureDumpWalletResult is a future promise to deliver the result of an
+// DumpWallet RPC invocation (or an applicable error)
+type FutureDumpWalletResult chan *response
+
+// Receive waits for the response promised by the future
+func (r FutureDumpWalletResult) Receive() error {
+	_, err := receiveFuture(r)
+	return err
+}
+
+// DumpWalletAsync returns an instance of a type that can be used to get the result
+// of the RPC at some future time by invoking the Receive function on the
+// returned instance.
+//
+// See DumpWalletAsync for the blocking version and more details.
+func (c *Client) DumpWalletAsync(destination string) FutureDumpWalletResult {
+	return c.sendCmd(btcjson.NewDumpWalletCmd(destination))
+}
+
+// DumpWallet dumps all wallet keys in a human-readable format to a server-side file.
+func (c *Client) DumpWallet(destination string) error {
+	return c.DumpWalletAsync(destination).Receive()
+}
+
+// FutureImportWalletResult is a future promise to deliver the result of an
+// ImportWalletAsync RPC invocation (or an applicable error)
+type FutureImportWalletResult chan *response
+
+// Receive waits for the response promised by the future
+func (r FutureImportWalletResult) Receive() error {
+	_, err := receiveFuture(r)
+	return err
+}
+
+// ImportWalletAsync returns an instance of a type that can be used to get the result
+// of the RPC at some future time by invoking the Receive function on the
+// returned instance.
+//
+// See ImportWallet for the blocking version and more details.
+func (c *Client) ImportWalletAsync(filename string) FutureImportWalletResult {
+	return c.sendCmd(btcjson.NewImportWalletCmd(filename))
+}
+
+// ImportWallet imports keys from a wallet dump file (see DumpWallet).
+func (c *Client) ImportWallet(filename string) error {
+	return c.ImportWalletAsync(filename).Receive()
+}
+
+// FutureUnloadWalletResult is a future promise to deliver the result of an
+// UnloadWalletAsync RPC invocation (or an applicable error)
+type FutureUnloadWalletResult chan *response
+
+// Receive waits for the response promised by the future
+func (r FutureUnloadWalletResult) Receive() error {
+	_, err := receiveFuture(r)
+	return err
+}
+
+// UnloadWalletAsync returns an instance of a type that can be used to get the result
+// of the RPC at some future time by invoking the Receive function on the
+// returned instance.
+//
+// See UnloadWallet for the blocking version and more details.
+func (c *Client) UnloadWalletAsync(walletName string) FutureUnloadWalletResult {
+	return c.sendCmd(btcjson.NewUnloadWalletCmd(walletName))
+}
+
+// UnloadWallet unloads the referenced wallet
+func (c *Client) UnloadWallet(walletName string) error {
```

Comment:
⚠️ `walletName` must be optional.

If the RPC server URL already contains the name of the wallet, like `http://127.0.0.1:8332/wallet/<walletname>`, it'll return an error.

From the docs:
> Specifying the wallet name on a wallet endpoint is invalid.